code/game/atoms.dm

```diff
@@ -42,6 +42,62 @@
 
 	var/admin_spawned = 0	//was this spawned by an admin? used for stat tracking stuff.
 
+	var/initialized = FALSE
+
+/atom/New(loc, ...)
+	if(use_preloader && (src.type == _preloader.target_path))//in case the instanciated atom is creating other atoms in New()
+		_preloader.load(src)
+	. = ..()
+	attempt_init(...)

```

Comment:
what about 
```
	var/list/created = SSatoms.created_atoms
	if(created)
		created += src
```?

Reply:
I have no idea what that tracker is for, I didn't port most of SSatoms. I figured it'd be a bad idea to add a new global list of all atoms without some use for it.

Reply:
Right, we'll wait until we get the entire thing ported over then.